### PR TITLE
Allow kotlin src.zip files to be passed into kotlinc

### PR DESCRIPTION
**Pull request by kageiit**
This allows passing in generated kotlin src zip files (from moshi/thrift etc.) similar to how javac already allows this

### CODE REVIEW

**Comment by kageiit** [comment]
cc @cwoodwar6 @thalescm

File: src/com/facebook/buck/jvm/kotlin/PathMatchers.java [hunk 1]
```diff
@@ -22,5 +22,7 @@
 
   public static final PathOrGlobMatcher KOTLIN_PATH_MATCHER = new PathOrGlobMatcher("**.kt");
 
+  public static final PathOrGlobMatcher SRC_ZIP_MATCHER = new PathOrGlobMatcher("**.src.zip");
```
Comment (cwoodwar6) [inline src/com/facebook/buck/jvm/kotlin/PathMatchers.java]: A test `src.zip` file would be nice as a reference that this is possible. Also helps ensure it never breaks.

Comment (kageiit) [reply src/com/facebook/buck/jvm/kotlin/PathMatchers.java]: Sure let me add a test

**Comment by kageiit** [comment]
@styurin added tests and rebased

**Comment by kageiit** [comment]
Test failures seem unrelated

**Comment by kageiit** [comment]
@styurin rebased and resolved conflicts. Thanks!

**Comment by kageiit** [comment]
@styurin rebased once more. Can we please merge? :)

**Pull request closed without merging**